File: Chapter0_Algorithm/2401/D240131/bj_25206.java
package D240131;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

public class bj_25206 {
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        double sum = 0.0; // 학점의 총합
        double total = 0.0; // 학점 × 과목평점

        Map<String, Double> map = new HashMap<>();
        map.put("A+", 4.5);
        map.put("A0", 4.0);
        map.put("B+", 3.5);
        map.put("B0", 3.0);
        map.put("C+", 2.5);
        map.put("C0", 2.0);
        map.put("D+", 1.5);
        map.put("D0", 1.0);

        for (int i = 0; i < 20; i++) {
            String[] subject = br.readLine().split(" ");

            Double grade = Double.parseDouble(subject[1]);

            if (subject[2].equals("P")) {
                continue;
            } else if (subject[2].equals("F")) {
                sum += grade;
            } else {
                // System.out.println("\n\n\n" + subject[2] + "\n");
                // System.out.println(map.get(subject[2]) + "dsdsds");
                total += (map.get(subject[2]) * grade);
                sum += grade;
            }

        }

        System.out.printf("%.6f", total / sum);
    }
}

File: Chapter0_Algorithm/2401/D240131/bj_25206.py
"""
인하대학교 컴퓨터공학과를 졸업하기 위해서는, 전공평점이 3.3 이상이거나 졸업고사를 통과해야 한다. 
그런데 아뿔싸, 치훈이는 깜빡하고 졸업고사를 응시하지 않았다는 사실을 깨달았다!

치훈이의 전공평점을 계산해주는 프로그램을 작성해보자.

전공평점은 전공과목별 (학점 × 과목평점)의 합을 학점의 총합으로 나눈 값이다.

인하대학교 컴퓨터공학과의 등급에 따른 과목평점은 다음 표와 같다.
"""

scores = {"A+": 4.5, "A0" : 4.0, "B+":3.5, "B0":3.0, "C+":2.5, "C0":2.0, "D+":1.5, "D0":1.0}

n = 20
sum = 0.0
total_credits = 0.0

for i in range(20):
    subject = input().split(" ")

    if subject[2] == "P":
        n -=1
    elif subject[2] == "F":
        total_credits += float(subject[1])
    else :
        total_credits += float(subject[1])
        a = scores[subject[2]]
        b = float(subject[1])
        sum+=(a*b)


# print(sum)
# print(total_credits)
# print(n)
result = sum / total_credits
print(round(result, 6))